### DISCUSSION TITLE
Is it possible to create multitable to one relationship with static field selector?

### DISCUSSION THREAD

**Discussion by mrspartak** [prisma/prisma#21589] [Q&A]
### Question

My idea was to store all translations in one table. Here is a simple example:
I see that it is only possible to connect field to field, not field to computed or field to static value.

```
model Translations {
  resource   String    @db.VarChar(16)
  
  language   String    @db.VarChar(2)
  key        String    @db.VarChar(32)
  value      String

  @@unique([resource, key, language], name: "translation_unique")
}

model Article {
  id             Int              @id @default(autoincrement())
  // title can be queried as Translation(resource=Article#<id>, language=en, key=title)
  // description as Translation(resource=Article#<id>, language=en, key=description)
}

model Channel {
  id             Int              @id @default(autoincrement())
  // name as Translation(resource=Channel#<id>, language=en, key=name)
}
```

**Comment by ludralph** [comment]
Hi @mrspartak 👋 

Thank you for raising this question.

Prisma does not currently support creating relations based on computed or static values. Relations in Prisma are defined between fields of models, and these fields correspond to columns in your database tables.

In your case, you're trying to create a relation between `Article` or `Channel` and `Translations` based on a computed value (resource=Article#<id>, resource=Channel#<id>). This is not currently supported in Prisma.

However, you can create a many-to-one relationship between `Article` or `Channel` and `Translations` by adding a foreign key field to the `Translations` model that references the `id` of the `Article` or `Channel` model. Here's an example:
```prisma
model Translations {
  id        Int      @id @default(autoincrement())
  resource  String   @db.VarChar(16)
  language  String   @db.VarChar(2)
  key       String   @db.VarChar(32)
  value     String
  articleId Int?
  channelId Int?
  article   Article? @relation(fields: [articleId], references: [id])
  channel   Channel? @relation(fields: [channelId], references: [id])
  @@unique([resource, key, language], name: "translation_unique")
}

model Article {
  id           Int            @id @default(autoincrement())
  translations Translations[]
}

model Channel {
  id           Int            @id @default(autoincrement())
  translations Translations[]
}
```

In this schema, each `Article` or `Channel` can have multiple associated `Translations` records. You would then need to ensure that the resource field in the `Translations` table is set to the correct value (Article#<id> or Channel#<id>) when creating a new translation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by mrspartak** [reply]
Thanks @ludralph. I thought about this approach, but because this is the main translation storage, I would have to create a foreign key for each column in each table I need to be translated into multiple languages. For now, I see that I need to write raw SQL or do many queries.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
